colibri/src/coingecko.rs

```diff
@@ -54,6 +58,50 @@ impl Coingecko {
             Err(e) => error!("Failed to query coingecko for {} due to {}", asset_id, e),
         }
 
+        debug!("Icon not found in coingecko for {}", asset_id);
         None
     }
 }
+
+#[cfg(test)]
```

Comment:
Hmmm I know this is possible in rust but it's ugly to have tests in the same file as implementation. Please use the separate tests directory approach.

- Create a tests directory at the same level as src. 
- Create test files there (e.g., tests/coingecko_test.rs)

And in that test you would use coingecko and test it

Reply:
As mentioned in discord the tests/ folder doesn't behave like this cfg test. That folder doesn't have access to private functions and also macros aren't accessible. There are a couple of details. The tests folder is meant to be used when testing libraries that is what I believe we should do. Implement as a library and then use it in the rotki binary. But we can see in the future, for reference https://doc.rust-lang.org/book/ch11-03-test-organization.html#integration-tests-for-binary-crates

I need to add tests for the api endpoints also and I'll do that in the tests folder